src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -72,6 +72,70 @@ public static bool IsScript(this SyntaxTree syntaxTree)
             return default;
         }
 
+        /// <summary>
+        /// Gets a token that contains or which start edge touches <paramref name="position"/>. I.e returns token 
+        /// directly to the right from <paramref name="position"/> or a token that encompasses <paramref name="position"/>.
+        /// If such Token doesn't exist a Kind = None Token is returned. 
+        /// </summary>
+        public static async Task<SyntaxToken> GetTouchingTokenToRightOrInAsync(
+            this SyntaxTree syntaxTree,
+            int position,
+            CancellationToken cancellationToken,
+            bool findInsideTrivia = false)
+        {
+            Contract.ThrowIfNull(syntaxTree);
+
+            if (position >= syntaxTree.Length)
+            {
+                return default;
+            }
+
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var token = root.FindToken(position, findInsideTrivia);
+            if (token.Span.Contains(position))
```

Comment:
i don't think we need a helper tha conveys just these two operations.  I think it would be just as clear to just do FindToken, but then do a .Contains.

Reply:
The idea was that for climbing the edges (discussed in "previous" PR (that's still not merged due to conflicts)) you need to identify on what edges are you. And you could theoretically be on edges of two Tokens (I actually wasn't able to come up with an example but thought it is possible). 

And thus you need to get a touching Token to the left and touching token to the right. If my assumption that two tokens could have touching .`Span` then I'll happily remove them and move back only to .GetTouchingToken.

Reply:
Superseded by discussion below.